### ISSUE TITLE
The loss function difference between this code and the original code

### ISSUE THREAD

**Issue by zxzStar** [TrsNium/Conditional_C_RNN_GAN#2]
Hi TrsNium,

Thanks for your code which makes me understand the Conditional GAN clearly. I appreciate it.

But I have a question about the loss function in this code:

In the model.py, you define the loss function is: 
    d_loss =  -tf.reduce_mean(d_r - d_f) 
    g_loss = -tf.reduce_mean(d_f). 
When I train the model, the loss can be the negative number, is it make sense?

In the original code provided by dhruvsharma1992, the loss function is 
    g_loss = tf.reduce_mean(-tf.log(tf.clip_by_value(self.generated_d, 1e-1000000, 1.0)))
    d_loss = tf.reduce_mean(-2*tf.log(tf.clip_by_value(self.real_d, 1e-1000000, 1.0)) 
                                 -tf.log(1 - tf.clip_by_value(self.generated_d, 0.0, 1.0-1e-1000000)) 
                                 -2*tf.log(1 - tf.clip_by_value(self.wrong_d, 0.0, 1.0-1e-1000000)))

Could you explain it?

Thank you very much!

**Comment by TrsNium** [comment]
Thanks for asking.

It's just my sense.

I tuning this model terms of loss function based on basic loss function such as cross-entropy but I cannot avoid  mode collapse.
And to avoid mode collapse , I tried to pre training generator.
But that didn't work too.

I am sorry I could not give useful information

**Comment by zxzStar** [comment]
Thank you for the reply. This code has already helped me to understand the conditional gan. I appreciate it. 

But I have a question.

The loss function you used is different from the original paper. So I wonder have you tried the same loss function as the original paper? If you have, is the original paper's loss function also meet the model collapse?

P.S. I want to use the Conditional C-RNN-GAN to do some classification. So the performance of Discriminator is more important for me. Do you have some suggestions?

Thank you very much.

**Comment by TrsNium** [comment]
If you have, is the original paper's loss function also meet the model collapse?
>I also tried it, and meet mode collapse

Do you have some suggestions?
> these days, generating image task is improved because Discriminator and Generator employed residual network and more useful loss function. So if we can employ residual in RNN, we might improve quality.

Thank you .

**Comment by zxzStar** [comment]
OK, now I know it. Thank you very much!
